apollo-router/src/services/router_service.rs

```diff
@@ -196,7 +196,7 @@ where
                         })
                     })
                     .unwrap_or_else(|| {
-                        Err(("missing query string", "missing query string".to_string()))
+                        Err(("Request missing query parameter in URL", "Request missing query parameter in URL".to_string()))
```

Comment:
I love the intention to improve this (and this is a great catch because this is not as helpful as it could be, as you noted), but I think this iteration still might lead users astray since it could lead them down the path of believing their GraphQL operation must be sent as a `GET` request when it's in fact more common to send it as a `POST` request?
 
Put another way, it's not **just** that there's a missing `query` parameter in URL, but it's that we didn't receive a `query` as either a `POST` or as a query-string parameter to a `GET` request.  And it's not necessarily ideal that they put that as a `query` parameter?

Just riffing on this a bit, what if this said:

```
There was no GraphQL operation to execute.  Use the `query` parameter to send an operation, using either GET or POST.
```

WDYT?

Reply:
Thanks for the suggestion. I think it will be more appropriate.

Reply:
I have updated the error message in the latest commit.